src/core/TESTING.md

```diff
@@ -62,8 +62,257 @@ test('my test', async () => {
+```ts
+router.post(
+  {
+    path: '/myPlugin/something',
+    validate: {
+      body: schema.object({
+        text: schema.string({ maxLength: 100 }),
+      }),
+    },
+  },
+  async (context, request, response) => {
+    const savedObjectsClient = context.core.savedObjects.client;
+    const { id } = await savedObjectsClient.update('myPlugin-type', 'myPlugin', {
+      something: request.body.text,
+    });
+    return response.ok({ body: { id } });
+  }
+);
+
+router.get(
+  {
+    path: '/myPlugin/something/{id}',
+    validate: {
+      params: schema.object({
+        id: schema.string(),
+      }),
+    },
+  },
+  async (context, request, response) => {
+    const savedObjectsClient = context.core.savedObjects.client;
+    const { attributes } = await savedObjectsClient.get('myPlugin-type', request.params.id);
+
+    return response.ok({
+      body: attributes.something,
+    });
+  }
+);
+```
+You can reuse existing [api_integration](/test/api_integration/config.js) setup by registering a test file within a [test loader](/test/api_integration/apis/index.js). More about the existing FTR setup in the [contribution guide](/CONTRIBUTING.md#running-specific-kibana-tests)
+
+The tests cover:
+- authenticated / non-authenticated user access (when applicable)
+```ts
+// TODO after https://github.com/elastic/kibana/pull/53208/
+```
+- request validation
+```ts
+export default function({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  describe('myPlugin', () => {
+    it('validate params before to store something', async () => {
+      const response = await supertest
+        .post('/myPlugin/something')
+        .set('content-type', 'application/json')
+        .send({ text: 'aaa'.repeat(100) })
+        .expect(400);
+
+      expect(response.body).to.have.property('message');
+    });
+  });
+```
+- the main logic of the plugin
+```ts
+export default function({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  describe('myPlugin', () => {
+    it('stores something', async () => {
+      const response = await supertest
+        .post('/myPlugin/something')
+        .set('content-type', 'application/json')
+        .send({ text: 'aaa' })
+        .expect(200);
+
+      expect(response.body).to.have.property('id');
+      expect(response.body.id).to.be.a('string');
+    });
+
+    it('retrieves something', async () => {
+      const { body } = await supertest
+        .post('/myPlugin/something')
+        .set('content-type', 'application/json')
+        .send({ text: 'bbb' })
+        .expect(200);
+
+      const response = await supertest.get(`/myPlugin/something/${body.id}`).expect(200);
+      expect(response.text).be('bbb');
+    });
+  });
+```
+
+##### TestUtils
+It can be utilized if your plugin doesn't interact with the elasticsearch server. Runs tests against real Kibana server instance.
```

Comment:
imho other important difference with FTR is that is actually programmatically creates a `Root` kibana object instead of starting Kibana from CLI as FTR does.

Reply:
Ok, I had it in a draft. I will put it back, thanks!